core/modules/filter/filter.admin.inc

```diff
@@ -287,35 +292,42 @@ function filter_admin_format_form($form, &$form_state, $format) {
   }
   $form['#filters'] = $filters;
 
-  // Filter status.
-  $form['filters']['status'] = array(
+  // Filter order (tabledrag).
+  $form['filters'] = array(
     '#type' => 'item',
-    '#title' => t('Enabled filters'),
-    '#prefix' => '<div id="filters-status-wrapper">',
-    '#suffix' => '</div>',
+    '#title' => t('Filter settings'),
+    '#theme' => 'filter_admin_format_filter_order',
+  );
+  $form['filters']['filters'] = array(
+    '#type' => 'value',
+    '#value' => $filters,
   );
   foreach ($all_filter_info as $name => $filter_info) {
-    $form['filters']['status'][$name] = array(
+    $configure_form = _filter_build_configure_form($form, $form_state, $name, $format);
+    $configure_link = !empty($configure_form) ? l(t('Configure'), '#', array('attributes' => array('class' => 'configure-link'), 'fragment' => 'configure')) : '';
+    
+    $form['filters'][$name]['status'] = array(
       '#type' => 'checkbox',
-      '#title' => $filter_info['title'],
       '#default_value' => $filters[$name]->status,
-      '#parents' => array('filters', $name, 'status'),
-      '#description' => $filter_info['description'],
-      '#weight' => $filter_info['weight'],
+      '#attributes' => array('class' => array('filter-status')),
     );
-  }
-
-  // Filter order (tabledrag).
-  $form['filters']['order'] = array(
-    '#type' => 'item',
-    '#title' => t('Filter processing order'),
-    '#theme' => 'filter_admin_format_filter_order',
-  );
-  foreach ($all_filter_info as $name => $filter_info) {
-    $form['filters']['order'][$name]['filter'] = array(
+    $form['filters'][$name]['configure_link'] = array(
+      '#markup' => $configure_link,
+    );
+    $form['filters'][$name]['filter'] = array(
+      '#theme' => 'filter_admin_format_filter_display',
+      '#type' => 'item',
+      '#prefix' => '<div class="filter-td">',
+      '#suffix' => '</div>',
+    );
+    $form['filters'][$name]['filter']['title'] = array(
       '#markup' => $filter_info['title'],
     );
-    $form['filters']['order'][$name]['weight'] = array(
+    $form['filters'][$name]['filter']['details'] = array(
+      '#markup' => isset($format_tips[$name]['tip']) ? $format_tips[$name]['tip'] : '',
+    );
+    $form['filters'][$name]['filter']['configure'] = $configure_form;
```

Comment:
Not a problem @docwilmot! We'll leave this one to pick up later. I'd like
to give this a shot if there's time before 1.2.0. Right now my focus is on
Link and Email fields and finishing up the WYSIWYG image handling.

On Tue, Aug 18, 2015 at 7:38 AM docwilmot <notifications@github.com> wrote:

> In core/modules/filter/filter.admin.inc
> <https://github.com/backdrop/backdrop/pull/1016#discussion_r37305657>:
>
> >        '#markup' => $filter_info['title'],
> >      );
> > -    $form['filters']['order'][$name]['weight'] = array(
> > +    $form['filters'][$name]['filter']['details'] = array(
> > +      '#markup' => isset($format_tips[$name]['tip']) ? $format_tips[$name]['tip'] : '',
> > +    );
> > +    $form['filters'][$name]['filter']['configure'] = $configure_form;
>
> I'm going to roll this back to a single draggable table without the
> "disabled" section, and remove all the js involved. Then I'll have to leave
> it there, because the Ajax form stuff will influence the rest of the
> process (dialog, validation, final $form_state structure for saving, etc)
> and I really am completely at a loss with Ajax.
>
> Sorry to bail out at this stage.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/backdrop/backdrop/pull/1016/files#r37305657>.
>
-- 

--
Sent from mobile